zpeclient.go

```diff
@@ -300,3 +322,34 @@ func (c *Cache) authorize(principal string, check AthenzAccessCheck) (bool, erro
 	}
 	return false, nil
 }
+
+// parseUpdateTime - read the last update time as string from configmap, store in the cache
+func (c *Cache) parseUpdateTime(configmap interface{}) error {
+	obj, ok := configmap.(*corev1.ConfigMap)
+	if !ok {
+		return fmt.Errorf("Unable to cast interface to config map object")
+	}
+	c.log.Println("latest contact time is: ", obj.Data["latest_contact"])
+	lastUpdateTime := strings.ReplaceAll(obj.Data["latest_contact"], `"`, "")
+	c.lastUpdate = lastUpdateTime
+	return nil
+}
+
+// checkUpdateTime - check the last update timestamp stored in the cache is less/more than 2 hours period
+func (c *Cache) checkUpdateTime() (res bool, err error) {
+	t, err := time.Parse(time.RFC3339Nano, c.lastUpdate)
+	if err != nil {
+		return false, fmt.Errorf("timestamp format in syncer config map is wrong")
+	}
+	t1 := time.Now()
+	t1.Format(time.RFC3339Nano)
+	diff := t1.Sub(t)
+
+	c.log.Println("time lapsed since last config map update: ", diff)
+	var max time.Duration = 2 * time.Hour
```

Comment:
could we make this duration configurable in `AuthorizationConfig`?

Reply:
Does this mean we will have a flag to configure this when auth-webhook starts?

Reply:
yep, we can have it ultimately defined through command line